efficientdet/keras/postprocess.py

```diff
@@ -354,20 +352,52 @@ def postprocess_per_class(params, cls_outputs, box_outputs, image_scales=None):
   return per_class_nms(params, boxes, scores, classes, image_scales)
 
 
-def generate_detections(params, cls_outputs, box_outputs, image_scales,
-                        image_ids):
-  """A legacy interface for generating [id, x, y, w, h, score, class]."""
+def generate_detections(params,
+                        cls_outputs,
+                        box_outputs,
+                        img_scales,
+                        img_ids,
+                        flip=False):
+  """A legacy interface for generating [id, x1, y1, x2, y2, score, class]."""
   nms_boxes_bs, nms_scores_bs, nms_classes_bs, _ = postprocess_per_class(
-      params, cls_outputs, box_outputs, image_scales)
-
-  image_ids_bs = tf.cast(tf.expand_dims(image_ids, -1), nms_scores_bs.dtype)
-  detections_bs = [
-      image_ids_bs * tf.ones_like(nms_scores_bs),
-      nms_boxes_bs[:, :, 1],
-      nms_boxes_bs[:, :, 0],
-      nms_boxes_bs[:, :, 3] - nms_boxes_bs[:, :, 1],
-      nms_boxes_bs[:, :, 2] - nms_boxes_bs[:, :, 0],
-      nms_scores_bs,
-      nms_classes_bs,
-  ]
+      params, cls_outputs, box_outputs, img_scales)
+
+  if flip:
+    _, width = utils.parse_image_size(params['image_size'])
+
+    img_ids_bs = tf.cast(tf.expand_dims(img_ids, -1), nms_scores_bs.dtype)
+    detections_bs = [
+        img_ids_bs * tf.ones_like(nms_scores_bs),
+        tf.expand_dims(img_scales, -1) * width - nms_boxes_bs[:, :, 3],
```

Comment:
I did not quite understand this logic. could you help clarify this? (maybe add some comments)?

Reply:
Extracted a variable and added comments - better?

Reply:
Thanks for adding this comment. Looks good.